services-core/src/gas_price/linear_interpolation.rs

```diff
@@ -0,0 +1,40 @@
+/// Linearly interpolate `value` between `points`.
+///
+/// `points` must not be empty and contain unique x values sorted in ascending order.
+/// If `value` is smaller than the first point or larger than the last it is clamped.
+pub fn interpolate(value: f64, points: &[(f64, f64)]) -> f64 {
+    assert!(!points.is_empty());
+    assert!(points.windows(2).all(|window| window[0].0 < window[1].0));
```

Comment:
Does this work with an array of size 1? Can we add a test for it?

Reply:
added